cli/tests/functional/pom.xml

```diff
@@ -89,23 +90,92 @@
             <version>${version.lib.helidon}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.helidon.build-tools.common</groupId>
+            <artifactId>helidon-build-common-test-utils</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>
         <plugins>
+            <!-- TODO: FIXME this is some hack to prevent maven-jar-plugin to create a jar twice and throw exception -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-jar</id>
+                        <phase/>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <systemPropertyVariables>
-                        <!-- version should be upgraded to 3.0.0 once released-->
-                        <helidon.test.version>2.4.0</helidon.test.version>
+                        <helidon.version>${project.version}</helidon.version>
                         <helidon.executable.directory>${project.basedir}/../../impl</helidon.executable.directory>
-                        <helidon.cli.version>3.0.0-M1</helidon.cli.version>
-                        <helidon.plugin.version>${project.version}</helidon.plugin.version>
+                        <io.helidon.build.common.maven.url.localRepo>${user.home}/.m2/repository</io.helidon.build.common.maven.url.localRepo>
                     </systemPropertyVariables>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>io.helidon.build-tools</groupId>
+                <artifactId>helidon-archetype-maven-plugin</artifactId>
+                <version>3.0.0-RC3</version>
```

Comment:
This creates a cycle. If we need to make changes to the plugin to fix a test we have to do a release.
It needs to work off the SNAPSHOT version. 

There are two ways to do this that I can see:
- use the maven invoker plugin to use the SNAPSHOT version of the maven plugin
- do what the plugin does with code within the tests.

The 1st one is cumbersome (as always).

The 2nd one might require a few changes to make some of the maven plugin code usable outside. (E.g. expose what you need so that you can re-use it, and have a way to implement the things that are plugin only ; e.g. the MavenArchiver stuff)

Reply:
I choose the first option !